src/bakabot/core/schedule/schedule.py

```diff
@@ -236,6 +239,22 @@ async def detect_changes(client: disnake.Client):
                 await ChangeDetector.changed_message(changed, client, schedulePair)
                 schedulePair.new.db_save()
 
+    @staticmethod
+    def handle_update_subjects_cache(schedules: tuple[Schedule, Schedule], client: InteractionBot):
+        """Updates the subjects cache with the subjects from the schedule"""
+
+        subjects = [
+            lesson.subject
+            for schedule in schedules
+            for day in schedule.days
+            for lesson in day.lessons
+            if lesson.subject
+        ]
+
+        hasMadeChanges = SubjectsCache.handleUpdateSubjects(subjects)
+        if hasMadeChanges:
+            SubjectsCache.updateCommandsWithSubjects(client)
```

Comment:
Basically the same as with the extraction from grades. The triple for loop to get all subjects is not really pretty but it gets the job done